src/main/java/manufacturing/dao/jdbc/CarDaoJdbcImpl.java

```diff
@@ -0,0 +1,215 @@
+            for (Driver driver : car.getDrivers()) {
+                preparedStatementForInserting.setLong(2, driver.getId());
+                preparedStatementForInserting.executeUpdate();
+            }
+            return car;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't update car in DB", e);
+        }
+    }
+
+    @Override
+    public boolean delete(Long carId) {
+        String query = "UPDATE cars "
+                + "SET deleted = TRUE "
+                + "WHERE car_id = ?";
+        int deletedID;
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, carId);
+            deletedID = preparedStatement.executeUpdate();
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't delete drivers by id " + carId, e);
+        }
+        return deletedID > 0;
+    }
+
+    @Override
+    public List<Car> getAll() {
+        String query = "SELECT c.car_id, c.car_model, "
+                + "c.manufacturer_id, m.manufacturer_name, m.manufacturer_country "
+                + "FROM cars c "
+                + "INNER JOIN manufacturers m "
+                + "ON c.manufacturer_id = m.manufacturer_id "
+                + "WHERE c.deleted = FALSE";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(createCar(resultSet, connection));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get all cars", e);
+        }
+    }
+
+    @Override
+    public List<Car> getAllByDriver(Long driverId) {
+        String query = "SELECT c.car_id, cd.driver_id "
+                + "FROM cars c "
+                + "INNER JOIN cars_drivers cd "
+                + "ON c.car_id = cd.car_id "
+                + "WHERE driver_id = ? AND deleted = FALSE";
+        List<Car> cars = new ArrayList<>();
+        try (Connection connection = ConnectionUtil.getConnection();
+                PreparedStatement preparedStatement = connection.prepareStatement(query)) {
+            preparedStatement.setLong(1, driverId);
+            ResultSet resultSet = preparedStatement.executeQuery();
+            while (resultSet.next()) {
+                cars.add(createCar(resultSet, connection));
+            }
+            return cars;
+        } catch (SQLException e) {
+            throw new DataProcessingException("Can't get Cars by Driver", e);
+        }
+    }
+
+    private List<Driver> getAllDriversByCarId(Long carId, Connection connection) {
```

Comment:
```suggestion
    private List<Driver> getAllDriversByCarId(Long carId) {
```
let's open a new connection in this method

Reply:
done